tiled/client/context.py

```diff
@@ -837,6 +837,25 @@ def create_service_principal(
             )
         ).json()
 
+    def delete_service_principal_apikey(self, uuid, first_eight=None):
```

Comment:
We have:

```py
c.context.create_api_key  # Create API key for currently-authenticated Principal
c.context.admin.create_api_key  # Create API key for any Principal (requires `admin:apikeys` scope)
```

The `admin` namespace is just an "accessor", a usability tweak to put methods that require admin scopes out of the path of ordinary users so that they do not clutter `c.context.<TAB>`. It's implemented as an `Admin` class defined in this module.

We also have:

```py
c.context.revoke_api_key  # Remove API key for currently-authenticated Principal
```

and I propose we add this method to the `Admin` class, similarly, name it `revoke_api_key`, so we have

```py
c.context.revoke_api_key  # Remove API key for currently-authenticated Principal
c.context.admin.revoke_api_key  # Remove API key for any Principal (requires `admin:apikeys` scope)
```

The docstring should also be broadened: this method and associated HTTP endpoint will in fact allow revoking _any_ API key. It is not restricted to Service Principals.

Reply:
Resolved in db0084ad67b89cfde25b2733e00980fbe321be27
The admin `revoke_api_key` command has been refactored for easier legibility